lib/enums/query/query_types.py

```diff
@@ -19,14 +22,25 @@ class QueryTypes(Enum):
     USER_QUERY = UserMapping
 
     @staticmethod
-    def from_string(val: str):
-        """
-        Converts a given string in a case-insensitive way to the enum values
-        """
-        try:
-            return QueryTypes[val.upper()]
-        except KeyError as err:
-            raise ParseQueryError(
-                f"Could not find query type {val}. "
-                f"Available query types are {','.join([prop.name for prop in QueryTypes])}"
-            ) from err
+    def _get_aliases() -> Dict:
+        return {
+            QueryTypes.FLAVOR_QUERY: [
+                "flavor",
+                "flavors",
+                "query_flavors",
+                "to_flavor_query",
+            ],
+            QueryTypes.PROJECT_QUERY: [
+                "project",
+                "projects",
+                "query_projects",
+                "to_project_query",
+            ],
+            QueryTypes.SERVER_QUERY: [
+                "server",
+                "servers",
+                "query_servers",
+                "to_server_query",
+            ],
+            QueryTypes.USER_QUERY: ["user", "users", "query_users", "to_user_query"],
```

Comment:
I don't think we should accept aliases for query objects, as the API will bloat rapidly with every combination

Anyone using an IDE will type `pro` and get every variation of project for example

Reply:
this is not for query objects - its specifically for when calling `then()` or `append_from()`. Importing queries stays the same

Reply:
But why do we need `then("query_servers", SOME_PROP, ...)` rather than just `then("servers")`

I agree on the plural but I don't think anybody has done `query_servers` in a then or append from for example?